webargs/pyramidparser.py

```diff
@@ -112,8 +115,10 @@ def wrapper(obj, *args, **kwargs):
                     request = req or obj.request
                 except AttributeError:  # first arg is request
                     request = obj
-                parsed_args = self.parse(schema, req=request, locations=locations,
-                                         validate=validate, force_all=as_kwargs)
+
+                parsed_args = self.parse(schema or argmap(request), req=request,
```

Comment:
I'm thinking this logic should go in `parse` rather than `use_args`. `parse`, `use_args`, and `use_kwargs` should all be able to take a schema factory as their first argument.